src/js/game/LevelMVC/LevelPlane.js

```diff
@@ -229,4 +232,83 @@ module.exports = class LevelPlane extends Array {
 
     return this[myIndex].blockType;
   }
+
+  getRedstone() {
+    this.redstoneList = [];
+    this.redstoneListON = [];
+    for (let i = 0; i < this.length; ++i) {
+      if (this[i].blockType.substring(0,12) === "redstoneWire") {
+        this[i].isPowered = false;
+        let position = [];
+        if (this.levelModel !== null) {
+          position = this.levelModel.controller.levelView.indexToCoordinates(i);
+        }
+        this.redstoneList.push(position);
+      }
+    }
+    for (let i = 0; i < this.length; ++i) {
+      if (this[i].blockType === "railsRedstoneTorch") {
+        let position = [];
+        if (this.levelModel !== null) {
+          position = this.levelModel.controller.levelView.indexToCoordinates(i);
+        }
+        this.redstonePropagation(position);
+      }
+    }
+
+    for (let obj in this.redstoneList) {
+      if (this.levelModel !== null) {
+        console.log("The block at index: " + this.redstoneList[obj] + " is a " + this[this.levelModel.controller.levelView.coordinatesToIndex(this.redstoneList[obj])].blockType + ". This block is NOT charged.");
+      }
+    }
+    for (let obj in this.redstoneListON) {
+      if (this.levelModel !== null) {
+        console.log("The block at index: " + this.redstoneListON[obj] + " is a " + this[this.levelModel.controller.levelView.coordinatesToIndex(this.redstoneListON[obj])].blockType + ". This block IS charged.");
+      }
+    }
+  }
+
+  redstonePropagation(position) {
+    let block = this[this.levelModel.yToIndex(position[1]) + position[0]];
+    //console.log("the position of " + block.blockType + " is " + "(" + position.x + ", " + position.y + "). This block is charged!");
```

Comment:
Not needed?

Reply:
I get that block because redstonePropagation is called on wire and batteries (torches or pressure plates) and I only do the list pushing and popping if it's wire, because the rest are just sort of "always charged"